sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -150,14 +187,34 @@ private static SpannerAccessor createAndConnect(SpannerConfig spannerConfig) {
     ValueProvider<String> host = spannerConfig.getHost();
     if (host != null) {
       builder.setHost(host.get());
+      instantiatingGrpcChannelProvider.setEndpoint(host.get());
     }
     ValueProvider<String> emulatorHost = spannerConfig.getEmulatorHost();
     if (emulatorHost != null) {
       builder.setEmulatorHost(emulatorHost.get());
       builder.setCredentials(NoCredentials.getInstance());
+    } else {
+      String userAgentString = USER_AGENT_PREFIX + "/" + ReleaseInfo.getReleaseInfo().getVersion();
+      /* Workaround to setup user-agent string.
+       * InstantiatingGrpcChannelProvider will override the settings provided.
+       * The section below and all associated artifacts will be removed once the bug
+       * that prevents setting user-agent is fixed.
+       * https://github.com/googleapis/java-spanner/pull/747
```

Comment:
Is this a link to a bug to be fixed (I see it's been merged already)?

If not, would you also add the link to the bug?

Reply:
- https://github.com/googleapis/java-spanner/pull/747 is the PR where the bug was introduced.
- https://github.com/googleapis/java-spanner/issues/880 is the issue.
- https://github.com/googleapis/java-spanner/pull/871 is the PR for the bug fix.

Reply:
Added https://github.com/googleapis/java-spanner/pull/871 in the comment.

Reply:
Thank you.